### PR TITLE
Tag cleanup and completion of tag exports

### LINKED ISSUE #39: Nested Datasets Need exposing to Python
Originally reported here #37.

See c++ example below
```
Dataset host;
Dataset held;
held.insert(Data::Value, "", {Dim::Row, 3}, {1, 2, 3});
host.insert(Data::Table, "", {}, {held});
```

### PR DESCRIPTION
**Pull request by SimonHeybrock**
- Remove data tags that encoded a data type --- the new way to achieve this is to use a custom `dtype` instead. Tags are not for types, but for conceptual grouping.
- Add Python exports for all dimensions and tags.

Also fixes #39. Nested datasets can be added as follows:

```python
        d = Dataset()
        table = Dataset()
        table[Data.Value, "col1"] = ([Dim.Row], [3.0,2.0,1.0,0.0])
        table[Data.Value, "col2"] = ([Dim.Row], np.arange(4.0))
        d[Data.Value, "data1"] = ([Dim.X], [table, table])
```

### CODE REVIEW

File: src/tags.h [hunk 1]
```diff
@@ -315,26 +316,25 @@ struct Data {
   using Value_t = detail::TagImpl<detail::DataDef::Value>;
   using Variance_t = detail::TagImpl<detail::DataDef::Variance>;
   using StdDev_t = detail::TagImpl<detail::DataDef::StdDev>;
-  using Int_t = detail::TagImpl<detail::DataDef::Int>;
-  using DimensionSize_t = detail::TagImpl<detail::DataDef::DimensionSize>;
-  using String_t = detail::TagImpl<detail::DataDef::String>;
+  // TODO Int and String is deprecated and should be removed, it is currently
+  // only required to maintain tests using MDZipView before it is properly
+  // refactored for multi-name support.
+  using DeprecatedInt_t = detail::TagImpl<detail::DataDef::Int>;
```
Comment (OwenArnold) [inline src/tags.h]: It's not clear to me why `Data::Int` was necessary is this simply to get a Tag with an underlying type of int given that it couldn't be derived from the data itself?

File: test/md_zip_view_test.cpp [hunk 2]
```diff
@@ -566,47 +567,48 @@ TEST(MDZipView, derived_standard_deviation) {
...
   ASSERT_EQ(it->get(Data::Value), 3.2);
-  ASSERT_EQ(it->get(Data::Int), 0);
+  ASSERT_EQ(it->get(Data::Variance), 0);
 }
 
 TEST(MDZipView, create_from_labels_with_name) {
   Dataset d;
   d.insert(Data::Value, "name", Dimensions(Dim::X, 2), 2);
-  d.insert(Data::Int, "name", {}, 1);
+  d.insert(Data::Variance, "name", {}, 1);
```
Comment (OwenArnold) [inline test/md_zip_view_test.cpp]: We already had the `Variance` tag - i must be misunderstanding something. Why did these tests use `Int` as the tag?

File: python/test/test_dataset.py [hunk 3]
```diff
@@ -112,6 +112,17 @@ def test_set_data(self):
...
+        d = Dataset()
+        table = Dataset()
+        table[Data.Value, "col1"] = ([Dim.Row], [3.0,2.0,1.0,0.0])
+        table[Data.Value, "col2"] = ([Dim.Row], np.arange(4.0))
+        d[Data.Value, "data1"] = ([Dim.X], [table, table])
+        d[Data.Value, "data1"].data[1][Data.Value, "col1"].data[0] = 0.0;
+        self.assertEqual(d[Data.Value, "data1"].data[0], table)
+        self.assertNotEqual(d[Data.Value, "data1"].data[1], table)
+        self.assertNotEqual(d[Data.Value, "data1"].data[0], d[Data.Value, "data1"].data[1])
+
```
Comment (OwenArnold) [inline python/test/test_dataset.py]: Good. Looks to be working exactly as I would expect.

File: src/tags.h (hunk 1 above)
Comment (SimonHeybrock) [reply src/tags.h]: Previously the type was defined by the tag, see we needed it as a separate tag. Now it is not necessary any more, except that we cannot actually remove it until `MDZipView` is refactored.

File: test/md_zip_view_test.cpp (hunk 2 above)
Comment (SimonHeybrock) [reply test/md_zip_view_test.cpp]: The actual tag in this test is irrelevant, as long as it is two different ones.

File: src/tags.h (hunk 1 above)
Comment (OwenArnold) [reply src/tags.h]: Ok. That's pretty much in line with what I understand now.

**Comment by OwenArnold** [comment]
@SimonHeybrock after conflicts resolved, happy to merge.

**Pull request merged**